File: sample/src/main/java/com/applandeo/materialcalendarsampleapp/MainActivity.java
package com.applandeo.materialcalendarsampleapp;

import android.content.Intent;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;

import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import com.annimon.stream.Stream;
import com.applandeo.materialcalendarview.CalendarUtils;
import com.applandeo.materialcalendarview.CalendarView;
import com.applandeo.materialcalendarview.DatePicker;
import com.applandeo.materialcalendarview.builders.DatePickerBuilder;
import com.applandeo.materialcalendarview.listeners.OnSelectDateListener;
import com.applandeo.materialcalendarview.utils.DateUtils;

import java.util.ArrayList;
import java.util.Calendar;
import java.util.List;

public class MainActivity extends AppCompatActivity implements OnSelectDateListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button openCalendarButton = (Button) findViewById(R.id.openCalendarButton);

        openCalendarButton.setOnClickListener(v -> {
            Intent intent = new Intent(this, CalendarActivity.class);
            startActivity(intent);
        });

        Button openOneDayPicker = (Button) findViewById(R.id.openOneDayPickerButton);
        openOneDayPicker.setOnClickListener(v -> startActivity(new Intent(this, OneDayPickerActivity.class)));

        Button openManyDaysPicker = (Button) findViewById(R.id.openManyDayPickerButton);
        openManyDaysPicker.setOnClickListener(v -> startActivity(new Intent(this, ManyDaysPickerActivity.class)));

        Button openRangePicker = (Button) findViewById(R.id.openRangePickerButton);
        openRangePicker.setOnClickListener(v -> startActivity(new Intent(this, RangePickerActivity.class)));

        Button openOneDayPickerDialog = (Button) findViewById(R.id.openOneDayPickerDialogButton);
        openOneDayPickerDialog.setOnClickListener(v -> openOneDayPicker());

        Button openManyDaysPickerDialog = (Button) findViewById(R.id.openManyDaysPickerDialogButton);
        openManyDaysPickerDialog.setOnClickListener(v -> openManyDaysPicker());

        Button openRangePickerDialog = (Button) findViewById(R.id.openRangePickerDialogButton);
        openRangePickerDialog.setOnClickListener(v -> openRangePicker());
    }

    private void openOneDayPicker() {
        Calendar min = Calendar.getInstance();
        min.add(Calendar.MONTH, -5);

        Calendar max = Calendar.getInstance();
        max.add(Calendar.DAY_OF_MONTH, 3);

        DatePickerBuilder oneDayBuilder = new DatePickerBuilder(this, this)
                .setPickerType(CalendarView.ONE_DAY_PICKER)
                .setDate(max)
                .setHeaderColor(R.color.colorPrimaryDark)
                .setHeaderLabelColor(R.color.currentMonthDayColor)
                .setSelectionColor(R.color.daysLabelColor)
                .setTodayLabelColor(R.color.colorAccent)
                .setDialogButtonsColor(android.R.color.holo_green_dark)
                .setDisabledDaysLabelsColor(android.R.color.holo_purple)
                .setPreviousButtonSrc(R.drawable.ic_chevron_left_black_24dp)
                .setForwardButtonSrc(R.drawable.ic_chevron_right_black_24dp)
                .setMinimumDate(min)
                .setMaximumDate(max)
                .setTodayColor(R.color.sampleLighter)
                .setHeaderVisibility(View.VISIBLE)
                .setDisabledDays(getDisabledDays());

        DatePicker oneDayPicker = oneDayBuilder.build();
        oneDayPicker.show();
    }

    private void openManyDaysPicker() {
        Calendar min = Calendar.getInstance();
        min.add(Calendar.DAY_OF_MONTH, -5);

        Calendar max = Calendar.getInstance();
        max.add(Calendar.DAY_OF_MONTH, 3);

        List<Calendar> selectedDays = new ArrayList<>(getDisabledDays());
        selectedDays.add(min);
        selectedDays.add(max);

        DatePickerBuilder manyDaysBuilder = new DatePickerBuilder(this, this)
                .setPickerType(CalendarView.MANY_DAYS_PICKER)
                .setHeaderColor(android.R.color.holo_green_dark)
                .setSelectionColor(android.R.color.holo_green_dark)
                .setTodayLabelColor(android.R.color.holo_green_dark)
                .setDialogButtonsColor(android.R.color.holo_green_dark)
                .setSelectedDays(selectedDays)
                .setNavigationVisibility(View.GONE)
                .setDisabledDays(getDisabledDays());

        DatePicker manyDaysPicker = manyDaysBuilder.build();
        manyDaysPicker.show();
    }

    private void openRangePicker() {
        Calendar min = Calendar.getInstance();
        min.add(Calendar.DAY_OF_MONTH, -5);

        Calendar max = Calendar.getInstance();
        max.add(Calendar.DAY_OF_MONTH, 3);

        List<Calendar> selectedDays = new ArrayList<>();
        selectedDays.add(min);
        selectedDays.addAll(CalendarUtils.getDatesRange(min, max));
        selectedDays.add(max);

        DatePickerBuilder rangeBuilder = new DatePickerBuilder(this, this)
                .setPickerType(CalendarView.RANGE_PICKER)
                .setHeaderColor(R.color.sampleDark)
                .setAbbreviationsBarColor(R.color.sampleLight)
                .setAbbreviationsLabelsColor(android.R.color.white)
                .setPagesColor(R.color.sampleLighter)
                .setSelectionColor(android.R.color.white)
                .setSelectionLabelColor(R.color.sampleDark)
                .setTodayLabelColor(R.color.dialogAccent)
                .setDialogButtonsColor(android.R.color.white)
                .setDaysLabelsColor(android.R.color.white)
                .setAnotherMonthsDaysLabelsColor(R.color.sampleLighter)
                .setSelectedDays(selectedDays)
                .setMaximumDaysRange(10)
                .setDisabledDays(getDisabledDays());

        DatePicker rangePicker = rangeBuilder.build();
        rangePicker.show();
    }

    private List<Calendar> getDisabledDays() {
        Calendar firstDisabled = DateUtils.getCalendar();
        firstDisabled.add(Calendar.DAY_OF_MONTH, 2);

        Calendar secondDisabled = DateUtils.getCalendar();
        secondDisabled.add(Calendar.DAY_OF_MONTH, 1);

        Calendar thirdDisabled = DateUtils.getCalendar();
        thirdDisabled.add(Calendar.DAY_OF_MONTH, 18);

        List<Calendar> calendars = new ArrayList<>();
        calendars.add(firstDisabled);
        calendars.add(secondDisabled);
        calendars.add(thirdDisabled);
        return calendars;
    }

    @Override
    public void onSelect(List<Calendar> calendars) {
        Stream.of(calendars).forEach(calendar ->
                Toast.makeText(getApplicationContext(),
                        calendar.getTime().toString(),
                        Toast.LENGTH_SHORT).show());
    }
}

File: sample/src/main/java/com/applandeo/materialcalendarsampleapp/MainActivity.kt
package com.applandeo.materialcalendarsampleapp

import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.applandeo.materialcalendarview.CalendarView
import com.applandeo.materialcalendarview.builders.DatePickerBuilder
import com.applandeo.materialcalendarview.getDatesRange
import com.applandeo.materialcalendarview.listeners.OnSelectDateListener
import com.applandeo.materialcalendarview.utils.getMidnightCalendar
import kotlinx.android.synthetic.main.activity_main.*
import java.util.*

class MainActivity : AppCompatActivity(), OnSelectDateListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        openCalendarButton.setOnClickListener {
            val intent = Intent(this, CalendarActivity::class.java)
            startActivity(intent)
        }

        openOneDayPickerButton.setOnClickListener { startActivity(Intent(this, OneDayPickerActivity::class.java)) }
        openManyDayPickerButton.setOnClickListener { startActivity(Intent(this, ManyDaysPickerActivity::class.java)) }
        openRangePickerButton.setOnClickListener { startActivity(Intent(this, RangePickerActivity::class.java)) }
        openOneDayPickerDialogButton.setOnClickListener { openOneDayPicker() }
        openManyDaysPickerDialogButton.setOnClickListener { openManyDaysPicker() }
        openRangePickerDialogButton.setOnClickListener { openRangePicker() }
    }

    private fun openOneDayPicker() {
        DatePickerBuilder(this, this).apply {
            pickerType(CalendarView.ONE_DAY_PICKER)
            date(Calendar.getInstance().apply { add(Calendar.DAY_OF_MONTH, 3) })
            headerColor(R.color.colorPrimaryDark)
            headerLabelColor(R.color.currentMonthDayColor)
            selectionColor(R.color.daysLabelColor)
            todayLabelColor(R.color.colorAccent)
            dialogButtonsColor(android.R.color.holo_green_dark)
            disabledDaysLabelsColor(android.R.color.holo_purple)
            previousButtonSrc(R.drawable.ic_chevron_left_black_24dp)
            forwardButtonSrc(R.drawable.ic_chevron_right_black_24dp)
            minimumDate(Calendar.getInstance().apply { add(Calendar.MONTH, -5) })
            maximumDate(Calendar.getInstance().apply { add(Calendar.DAY_OF_MONTH, 3) })
            disabledDays(disabledDays())
        }.build().show()
    }

    private fun openManyDaysPicker() {
        val selectedDays = mutableListOf<Calendar>().apply { addAll(disabledDays()) }
        selectedDays.add(Calendar.getInstance().apply { add(Calendar.DAY_OF_MONTH, -5) })
        selectedDays.add(Calendar.getInstance().apply { add(Calendar.DAY_OF_MONTH, 3) })

        DatePickerBuilder(this, this).apply {
            pickerType(CalendarView.MANY_DAYS_PICKER)
            headerColor(android.R.color.holo_green_dark)
            selectionColor(android.R.color.holo_green_dark)
            todayLabelColor(android.R.color.holo_green_dark)
            dialogButtonsColor(android.R.color.holo_green_dark)
            selectedDays(selectedDays)
            disabledDays(disabledDays())
        }.build().show()
    }

    private fun openRangePicker() {
        val min = Calendar.getInstance().apply { add(Calendar.DAY_OF_MONTH, -5) }
        val max = Calendar.getInstance().apply { add(Calendar.DAY_OF_MONTH, 3) }

        val selectedDays = mutableListOf<Calendar>()
        selectedDays.add(min)
        selectedDays.addAll(min.getDatesRange(max))
        selectedDays.add(max)

        DatePickerBuilder(this, this).apply {
            pickerType(CalendarView.RANGE_PICKER)
            headerColor(R.color.sampleDark)
            abbreviationsBarColor(R.color.sampleLight)
            abbreviationsLabelsColor(android.R.color.white)
            pagesColor(R.color.sampleLighter)
            selectionColor(android.R.color.white)
            selectionLabelColor(R.color.sampleDark)
            todayLabelColor(R.color.dialogAccent)
            dialogButtonsColor(android.R.color.white)
            daysLabelsColor(android.R.color.white)
            anotherMonthsDaysLabelsColor(R.color.sampleLighter)
            selectedDays(selectedDays)
            disabledDays(disabledDays())
        }.build().show()
    }


    private fun disabledDays(): List<Calendar> {
        val disabledDays = mutableListOf<Calendar>()

        disabledDays.add(getMidnightCalendar.apply { add(Calendar.DAY_OF_MONTH, 2) })
        disabledDays.add(getMidnightCalendar.apply { add(Calendar.DAY_OF_MONTH, 1) })
        disabledDays.add(getMidnightCalendar.apply { add(Calendar.DAY_OF_MONTH, 18) })
        return disabledDays
    }

    override fun onSelect(calendar: List<Calendar>) {
        calendar.forEach { day ->
            Toast.makeText(applicationContext, day.time.toString(), Toast.LENGTH_SHORT).show()
        }
    }
}


